cabal-install/tests/UnitTests/Distribution/Client/ProjectConfig.hs

```diff
@@ -248,6 +249,12 @@ prop_roundtrip_printparse_RelaxedDep rdep =
       [rdep'] -> rdep' == rdep
       _       -> False
 
+prop_roundtrip_printparse_RelaxDeps :: RelaxDeps -> Bool
+prop_roundtrip_printparse_RelaxDeps rdep =
+    case [ x | (x,"") <- Parse.readP_to_S Text.parse (Text.display rdep) ] of
```

Comment:
This never generates the `*`-variants, correct? May be worth introducing a helper function that does that.